src/renderer/components/Login.js

```diff
@@ -10,20 +10,21 @@ const {
   Button,
   InputGroup,
   FormGroup,
-  Collapse
+  Collapse,
+  Tooltip,
+  Intent
 } = require('@blueprintjs/core')
 
 class Login extends React.Component {
   constructor (props) {
     super(props)
-    this.state = this._defaultState(false)
+    this.state = this._defaultState(false, false, false)
 
     this.handleChange = this.handleChange.bind(this)
     this.handleSubmit = this.handleSubmit.bind(this)
-    this.handleShowAdvanced = this.handleShowAdvanced.bind(this)
   }
 
-  _defaultState (showAdvanced) {
+  _defaultState (showAdvanced, showPasswordMail, showPasswordSend) {
```

Comment:
instead of this, can _defaultState take an `opts` parameter which will override the default options using Object.assign?

ex: 
`
_defaultState (opts) {
  return Object.assign(theBigStateObject, opts || {})
}